sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -183,6 +183,266 @@ class CollationSuite extends DatasourceV2SQLBase {
     }
   }
 
+  test("checkCollation throws exception for incompatible collationIds") {
+    val left: String = "abc" // collate with 'UNICODE_CI'
+    val leftCollationName: String = "UNICODE_CI";
+    var right: String = null // collate with 'UNICODE'
+    val rightCollationName: String = "UNICODE";
+    // contains
+    right = left.substring(1, 2);
+    checkError(
+      exception = intercept[SparkException] {
+        spark.sql(s"SELECT contains(collate('$left', '$leftCollationName')," +
+          s"collate('$right', '$rightCollationName'))").collect()
```

Comment:
```suggestion
          s"collate('$right', '$rightCollationName'))")
```

Reply:
I think I need .collect() in this case, otherwise it doesn't produce an exception here

Reply:
added a check in alaysis as well, so this is finally removed